fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java

```diff
@@ -436,25 +419,82 @@ public void buildDatabaseSpecificArtifactsPostgres(PhysicalDataModel model) {
+                                .buildColumns();
+
+                        statements.add(new AddColumn(schemaName, tableName, cols.get(0)));
+                        statements.add(new AddColumn(schemaName, tableName, cols.get(1)));
+                        
+                        // Add the new index on REINDEX_TSTAMP. This index is special because it's the
+                        // first index in our schema to use DESC.
+                        final String mtId = this.multitenant ? MT_ID : null;
+                        List<OrderedColumnDef> indexCols = Arrays.asList(new OrderedColumnDef(REINDEX_TSTAMP, OrderedColumnDef.Direction.DESC, null));
+                        statements.add(new CreateIndexStatement(schemaName, IDX_LOGICAL_RESOURCES_RITS, tableName, mtId, indexCols));
+                    }
+                    return statements;
+                })
                 .build(pdm);
 
         // TODO should not need to add as a table and an object. Get the table to add itself?
         tbl.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
         this.procedureDependencies.add(tbl);
         pdm.addTable(tbl);
         pdm.addObject(tbl);
+        
+        // For V0006 we also add a couple of new columns and an index to support
+        // reindexing of resources. Note CURRENT_TIMESTAMP works for Derby, Postgres and Db2.
+//        List<ColumnBase> cols = new ColumnDefBuilder()
+//                .addTimestampColumn(REINDEX_TSTAMP, false, "CURRENT_TIMESTAMP")
+//                .addBigIntColumn(REINDEX_TXID, false, "0")
+//                .buildColumns();
+//        AlterTableAddColumn addCols = new AlterTableAddColumn(schemaName, tableName, FhirSchemaVersion.V0006.vid(), cols);
+//        addCols.addDependency(tbl); // table must be created before we try to alter it
+//        pdm.addObject(addCols);
+
+        // Make sure we have an index on the REINDEX_TSTAMP column so that we can quickly
+        // identify which resources need to be reindexed. For the reindex resource selection
+        // query, it's essential that we collate the reindex_tstamp with NULLS FIRST
+//        CreateIndex tsidx = CreateIndex.builder()
+//                .setSchemaName(schemaName)
+//                .setTableName(tableName)
+//                .setTenantColumnName(MT_ID)
+//                .setIndexName("IDX_" + LOGICAL_RESOURCES + "_RITS")
+//                .setVersion(FhirSchemaVersion.V0006.vid())
+//                .addColumn(REINDEX_TSTAMP, OrderedColumnDef.Direction.DESC, null)
+//                .build();
+//        tsidx.addDependency(addCols);
+//        pdm.addObject(tsidx);
+
+
+        // Create a new sequence to use as a transaction id for our reindexing process
+//        Sequence seq = new Sequence(schemaName, FhirSchemaConstants.REINDEX_SEQ, FhirSchemaVersion.V0001.vid(), 1, 100, 1);
+//        seq.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
+//        sequencePrivileges.forEach(p -> p.addToObject(seq));
+//        pdm.addObject(seq);
```

Comment:
is this useful?

Reply:
Cruft left over from the alter table bug in Derby. Cleaned up.